### PR TITLE
Updated soar for interface changes

### PR DESCRIPTION
**Pull request by dmcollom**

### CODE REVIEW

File: tom_observations/facilities/soar.py [hunk 1]
```diff
@@ -109,9 +113,7 @@ class SOARFacility(LCOFacility):
...
     def get_form(self, observation_type):
-        if observation_type == 'IMAGING':
-            return SOARImagingObservationForm
-        elif observation_type == 'SPECTRA':
-            return SOARSpectroscopyObservationForm
-        else:
+        try:
+            return self.observation_forms[observation_type]
+        except KeyError:
             return SOARBaseObservationForm
```
Comment (phycodurus) [inline tom_observations/facilities/soar.py:119]: this indentation has me worried

Comment (phycodurus) [reply tom_observations/facilities/soar.py:119]: I guess if `SOARBaseObservationForm` is what you want to return on `KeyError` , then why not `return self.observation_forms.get(observation_type, SOARBaseObservationForm)` ?

**Review by phycodurus — APPROVED** [review]

File: tom_observations/facilities/soar.py (hunk 1 above)
Comment (dmcollom) [reply tom_observations/facilities/soar.py]: done

File: tom_observations/facilities/lco.py [hunk 2]
```diff
@@ -803,10 +803,7 @@ class LCOFacility(BaseRoboticObservationFacility):
     }
 
     def get_form(self, observation_type):
-        try:
-            return self.observation_forms[observation_type]
-        except KeyError:
-            return LCOBaseObservationForm
+        return self.observation_forms.get('observation_type', LCOBaseObservationForm)
```
Comment (phycodurus) [inline tom_observations/facilities/lco.py:806]: I don't think you want the string-literal in the get.

File: tom_observations/facilities/soar.py [hunk 3]
```diff
@@ -109,9 +113,4 @@ class SOARFacility(LCOFacility):
     }
 
     def get_form(self, observation_type):
-        if observation_type == 'IMAGING':
-            return SOARImagingObservationForm
-        elif observation_type == 'SPECTRA':
-            return SOARSpectroscopyObservationForm
-        else:
-            return SOARBaseObservationForm
+        return self.observation_forms.get('observation_type', SOARBaseObservationForm)
```
Comment (phycodurus) [inline tom_observations/facilities/soar.py:116]: again, you don't want the string-literal in the get, right?

**Review by phycodurus — CHANGES_REQUESTED** [review]

**Review by phycodurus — APPROVED** [review]

**Pull request merged**